rotkehlchen/data_import/importers/bitcoin_tax.py

```diff
@@ -0,0 +1,337 @@
+            self,
+            cursor: DBCursor,
+            csv_row: dict[str, Any],
+            event_identifier: str,
+            timestamp: TimestampMS,
+            location: Location,
+            action: str,
+            asset: AssetWithOracles,
+            asset_amount: AssetAmount,
+            fee_asset: Optional[AssetWithOracles],
+            fee_amount: Fee,
+            memo: str,
+    ) -> None:
+        try:
+            event_type, event_subtype = self.ACTION_TO_HISTORY_EVENT_TYPE[action]
+        except KeyError as e:
+            raise UnsupportedCSVEntry(
+                f'Unsupported entry {action}. Data: {csv_row}') from e
+
+        events: list[HistoryBaseEntry] = []
+        event = HistoryEvent(
+            event_identifier=event_identifier,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=location,
+            asset=asset,
+            balance=Balance(
+                amount=asset_amount,
+                usd_value=ZERO,
+            ),
+            notes=memo,
+            event_type=event_type,
+            event_subtype=event_subtype,
+        )
+        events.append(event)
+        if fee_amount != ZERO and fee_asset is not None:
+            fee_event = HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=1,
+                timestamp=timestamp,
+                location=location,
+                asset=fee_asset,
+                balance=Balance(
+                    amount=fee_amount,
+                    usd_value=ZERO,
+                ),
+                notes=memo,
+                event_type=HistoryEventType.SPEND,
+                event_subtype=HistoryEventSubType.FEE,
+            )
+            events.append(fee_event)
+        self.add_history_events(cursor, events)
+
+    def _consume_event(
+            self,
+            cursor: DBCursor,
+            csv_row: dict[str, Any],
+            csv_type: str,
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S %z',
+    ) -> None:
+        """
+        Consume the file containing trades from BitcoinTax.
+        - UnknownAsset
+        - DeserializationError
+        """
+        if all(value == '' for value in csv_row.values()):
+            return  # skip empty rows
+
+        # use a deterministic event_identifier to avoid duplicate events in case of reimport
+        event_identifier = f'rotki_events_{self.hash_csv_row(csv_row)}'
+        timestamp = TimestampMS(deserialize_timestamp_from_date(
+            date=csv_row['Date'],
+            formatstr=timestamp_format,
+            location='BitcoinTax',
+        ) * 1000)
+
+        try:
+            location = Location.deserialize(csv_row['Account'])
+        except DeserializationError:
+            location = self.csv_value_to_location(csv_row['Account'])
+
+        asset_resolver = LOCATION_TO_ASSET_MAPPING.get(location, asset_from_common_names)
+        base_asset = asset_resolver(csv_row['Symbol'])
```

Comment:
I think it will make sense to use `AssetBalance` here in order to reduce the number of variables used.

Additionally for fees I think it will make sense to do the check `fee_amount != ZERO and fee_asset is not None` here and use `None` if there is no fee.